internal/core/command/restDevice_test.go

```diff
@@ -0,0 +1,395 @@
+	Name:       NAME,
+	AdminState: models.Locked,
+	Service: models.DeviceService{
+		Addressable: models.Addressable{Protocol: "http", Address: "localhost",
+			Port: 8080}}}
+
+type mockOutline struct {
+	methodName string
+	arg        []interface{}
+	ret        []interface{}
+}
+
+func TestRestGetCommandsByDeviceName(t *testing.T) {
+	tests := []struct {
+		name           string
+		dcMock         *mocks.DeviceClient
+		dbMock         interfaces.DBClient
+		request        *http.Request
+		expectedErr    error
+		expectedStatus int
+	}{
+		{
+			"ok json content type in header",
+			createMockUnlockedDeviceCommandClient(deviceId),
+			createMockWithOutlines([]mockOutline{
+				{"GetCommandsByName", []interface{}{knownDeviceName}, []interface{}{nil, nil}},
+				{"GetCommandsByDeviceId", []interface{}{deviceId}, []interface{}{[]models.Command{exampleCommand}, nil}},
+			}),
+			createRequestWithPathParameters(http.MethodGet, map[string]string{ID: deviceId, COMMANDID: exampleCommand.Id, clients.ContentType: clients.ContentTypeJSON}, exampleCommand, clients.ContentTypeJSON),
+			nil,
+			http.StatusOK,
+		},
+		{
+			"ok cbor content type in header",
+			createMockUnlockedDeviceCommandClient(deviceId),
+			createMockWithOutlines([]mockOutline{
+				{"GetCommandsByName", []interface{}{knownDeviceName}, []interface{}{nil, nil}},
+				{"GetCommandsByDeviceId", []interface{}{deviceId}, []interface{}{[]models.Command{exampleCommand}, nil}},
+			}),
+			createRequestWithPathParameters(http.MethodGet, map[string]string{ID: deviceId, COMMANDID: exampleCommand.Id, clients.ContentType: clients.ContentTypeJSON}, exampleCommand, clients.ContentTypeCBOR),
+			nil,
+			http.StatusOK,
+		},
+		{
+			"no device for requested name",
+			createMockUnlockedDeviceCommandErrorClient(deviceName),
+			createMockWithOutlines([]mockOutline{
+				{"GetCommandsByName", []interface{}{unknownDeviceName}, []interface{}{models.DeviceService{}, nil}},
+				{"GetCommandsByDeviceId", []interface{}{deviceId}, []interface{}{[]models.Command{}, nil}},
+			}),
+			createGetCommandRequest(),
+			nil,
+			http.StatusNotFound,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			rr := httptest.NewRecorder()
+			var loggerMock = logger.NewMockClient()
+			configuration := config.ConfigurationStruct{
+				Service: bootstrapConfig.ServiceInfo{MaxResultCount: 1},
+			}
+			restGetCommandsByDeviceName(
+				rr,
+				tt.request,
+				tt.dbMock,
+				tt.dcMock,
+				&configuration,
+				errorconcept.NewErrorHandler(loggerMock))
+			response := rr.Result()
+			if response.StatusCode != tt.expectedStatus {
+				t.Errorf("status code mismatch -- expected %v got %v", tt.expectedStatus, response.StatusCode)
+				return
+			}
+		})
+	}
+}
+
+func TestRestPutDeviceCommandByCommandID(t *testing.T) {
+	tests := []struct {
+		name           string
+		dcMock         *mocks.DeviceClient
+		dbMock         interfaces.DBClient
+		request        *http.Request
```

Comment:
This test argument is always `nil`. Can we actually leverage this and pass a request with different headers and assert the request sent to the device service has the correct headers? As I see it we are not really testing the newly added functionality and not verifying that the `Content-type` header is present. I think we need those tests as well as negative tests which ensure we are not passing additional unwanted headers.